novawallet/Common/Services/CallFormatting/CallFormattingOperationFactory.swift

```diff
@@ -225,48 +233,104 @@ private extension CallFormattingOperationFactory {
             decoded: decodedCall
         )
     }
+
+    func createDecodingWrapper(
+        dependingOn runtimeCodingServiceWrapper: CompoundOperationWrapper<RuntimeCodingServiceProtocol>,
+        for callData: Substrate.CallData,
+        chainId _: ChainModel.Id
+    ) -> CompoundOperationWrapper<JSON> {
+        let decodingWrapper: CompoundOperationWrapper<JSON> = OperationCombiningService.compoundNonOptionalWrapper(
+            operationQueue: operationQueue
+        ) {
+            let codingService = try runtimeCodingServiceWrapper.targetOperation.extractNoCancellableResultData()
+
+            return codingService.createDecodingWrapper(
+                for: callData,
+                of: GenericType.call.name
+            )
+        }
+
+        return decodingWrapper
+    }
+
+    func createCodingFactoryWrapper(
```

Comment:
I think we could just a add to the extension of RuntimeCodingServiceProviderProtocol a method of getting wrapper with RuntimeCoderFactoryProtocol value directly